src/market.rs

```diff
@@ -16,19 +19,36 @@ impl Market {
     }
 
     pub async fn fetch_items(&self) -> Result<ItemsApiResponse> {
-        let url = BASE_URL.join("/items")?;
-        let resp = self.client.get(url).send().await?;
-        Ok(resp.json())
+        Ok(self.client.get(ITEMS).send().await?.json().await?)
     }
 
-
     pub async fn fetch_orders(&self, item_url: String) -> Result<OrdersApiResponse> {
-        let url = BASE_URL.join(&format!("/items/{}/orders", item_url))?;
-        let resp = self.client.get(url).send().await?;
-        let body = resp.text().await?;
-        let json_resp: OrdersApiResponse = serde_json::from_str(&body)?;
-        Ok(json_resp)
+        let url = format!("{ITEMS}/{}/orders", item_url);
+        Ok(self.client.get(&url).send().await?.json().await?)
     }
+
```

Comment:
What about SOLID principle?

Reply:
What is SOLID. Rust is syntax sugar above Haskell